scipy/stats/_multivariate.py

```diff
@@ -4625,7 +4626,7 @@ def asymptotic(dim, df):
 
         # preserves ~12 digits accuracy up to at least `dim=1e5`. See gh-18465.
         threshold = dim * 100 * 4 / (np.log(dim) + 1)
-        return _lazywhere(df >= threshold, (dim, df), f=asymptotic, f2=regular)
+        return xpx.apply_where(df < threshold, regular, asymptotic, (dim, df))
```

Comment:
I'm not sure if it matters here, but reversing the order can change what happens with NaNs. Was they idea that the "regular" case should come first?

Reply:
I didn't consider the use case of NaNs. Should I revert all these swaps just to be safe?

Reply:
It would be safer, but I think whether to swap depends on whether there is going to be any asymmetry between the arguments. For instance, in `np.testing.assert_close`, the first argument should be the actual result, and second should be the (desired) reference value. Is there any such concept with `apply_where`? If not, fine to change it back.

Reply:
> Is there any such concept with apply_where?

`apply_where(cond, f1, f2, args)` will always produce identical results to `apply_where(~cond, f2, f1, args)`.
There isn't any performance difference either.

Reply:
K, then yeah, I suppose it's safer to change it back.

Reply:
Reverted all swaps